File: services/api/app/schemas/couple_private.py
"""Schemas: privater, flankierender Echo-Dialog zur Paarsitzung.

Diese Inhalte verlassen den Server ausschließlich an die Person, der sie gehören.
Siehe services/couple_private_service.py.
"""
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class CouplePrivateMessageCreate(BaseModel):
    content: str = Field(..., min_length=1, max_length=4000)


class CouplePrivateMessage(BaseModel):
    id: UUID
    role: str      # 'user' | 'echo'
    kind: str      # 'chat' | 'feedback'
    content: str
    #: 'acute' | 'elevated', wenn die Sicherheits-Triage eingegriffen hat.
    safety: str | None = None
    created_at: datetime


class CouplePrivateThread(BaseModel):
    messages: list[CouplePrivateMessage]

File: apps/web/src/api/couplePrivate.ts
import { apiClient } from './client'

/**
 * Privater, flankierender Echo-Dialog zur Paarsitzung.
 *
 * Diese Inhalte gehören ausschließlich der angemeldeten Person – die Partnerperson hat
 * keinen Endpunkt, über den sie sie sehen könnte.
 */
export interface CouplePrivateMessage {
  id: string
  role: 'user' | 'echo'
  kind: 'chat' | 'feedback'
  content: string
  /** 'acute' | 'elevated', wenn die Sicherheits-Triage eingegriffen hat. */
  safety?: 'acute' | 'elevated' | null
  created_at: string
}

export interface CouplePrivateThread {
  messages: CouplePrivateMessage[]
}

export const couplePrivateApi = {
  get: (sessionId: string) =>
    apiClient.get<CouplePrivateThread>(`/couple/sessions/${sessionId}/private`).then(r => r.data),

  send: (sessionId: string, content: string) =>
    apiClient
      .post<CouplePrivateThread>(`/couple/sessions/${sessionId}/private`, { content })
      .then(r => r.data),

  feedback: (sessionId: string) =>
    apiClient
      .post<CouplePrivateThread>(`/couple/sessions/${sessionId}/private/feedback`)
      .then(r => r.data),
}
